app/auto-eligibility/register-your-interest/message-receiver.js

```diff
@@ -1,25 +1,40 @@
 const { MessageReceiver } = require('ffc-messaging')
 const registerYourInterestConfig = require('../../config').registerYourInterestConfig
-const processRegisterYourInterestRequest = require('./process-register-your-interest')
+const telemetryClient = require('../../app-insights/telemetry-client')
+const processRegisterYourInterest = require('./process-register-your-interest')
 
-let registerYourInterestReceiver
+let messageReceiver
 
 const start = async () => {
   try {
-    const registerYourInterestMessageHandler = message => {
-      processRegisterYourInterestRequest(message.body)
-      registerYourInterestReceiver.completeMessage(message)
-    }
-    registerYourInterestReceiver = new MessageReceiver(registerYourInterestConfig.registerYourInterestRequestQueue, registerYourInterestMessageHandler)
-    await registerYourInterestReceiver.subscribe()
-    console.info(`${new Date().toISOString()} Ready to receive messages...`)
-  } catch (e) {
-    console.error(`${new Date().toISOString()} Error starting message receiver`, e)
+    messageReceiver = new MessageReceiver(
+      registerYourInterestConfig.registerYourInterestRequestQueue,
+      message => {
+        try {
+          processRegisterYourInterest(message.body)
+          messageReceiver.completeMessage(message)
+          console.log(`${new Date().toISOString()} Register your interest message has been processed`)
+        } catch (error) {
+          messageReceiver.deadLetterMessage(message)
```

Comment:
FYI @marcin-mogiela  / @json-hamilton raising a technical debt ticket to test this closure and specifically the dead letter queue function call. Appreciate it is tricky right now and needs more time.

https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/148589